### ISSUE TITLE
Implement (debug) logging framework with options to write to file or console

### ISSUE THREAD

**Issue by JoostvanPinxten** [kees-jan/scroom#63]
Instead of writing to the console through printf or std::cout calls, I believe the SCROOM GUI app should use a logging framework to write messages that are to be used either for understanding implementation issues, or for debugging what is going wrong on file load etc.

The [spdlog C++ library](https://github.com/gabime/spdlog)  sounds like a promising library to achieve just this.

**Comment by kees-jan** [comment]
I agree. However, before we can do that, we need to integrate with a package manager. Ubuntu 18.04 currently ships with spdlog 0.16.3, which is a bit old.

However, before we do that, I'd like to complete the switch to cmake. I've found an example CMakefile that integrates with conan, which is a promising package manager (see branch "thuis" for the current state of this).

So plenty of work remaining :wink:
